setup.py

```diff
@@ -65,7 +65,7 @@
     install_requires=[
         "trio >= 0.22.0",
         "outcome",
-        "sniffio",
+        "sniffio >= 1.3.0",
```

Comment:
When I was thinking about how I'd do this a few days ago I thought about bumping the `trio` dep too. Not sure if you want to do that though.

Reply:
I tested the new change against both Trio 0.22 and Trio 0.23. Since the thread-local wins, and trio-asyncio's setting of it is more tightly scoped than Trio's, the new trio-asyncio logic works fine with either behavior on the Trio side.